Source/Core/DiscIO/NANDImporter.cpp

```diff
@@ -0,0 +1,164 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DiscIO/NANDImporter.h"
+
+#include "Common/Crypto/AES.h"
+#include "Common/MsgHandler.h"
+#include "Common/Swap.h"
+#include "DiscIO/NANDContentLoader.h"
+
+namespace DiscIO
+{
+void NANDImporter::ImportNANDBin(const std::string& path_to_bin)
+{
+  if (!ReadNANDBin(path_to_bin))
+    return;
+  FindSuper();
+  const std::string nand_root = File::GetUserPath(D_WIIROOT_IDX);
+  ProcessEntry(0, nand_root);
+  ExportKeys(nand_root);
+  ExtractCertificates(nand_root);
+
+  // We have to clear the cache so the new NAND takes effect
+  DiscIO::CNANDContentManager::Access().ClearCache();
+}
+
+bool NANDImporter::ReadNANDBin(const std::string& path_to_bin)
+{
+  File::IOFile file(path_to_bin, "rb");
+  if (file.GetSize() != NAND_BIN_SIZE)
+  {
+    PanicAlertT("This file does not look like a BootMii NAND backup. (0x%x does not equal 0x%x.)",
+                file.GetSize(), NAND_BIN_SIZE);
+    return false;
+  }
+  m_nand = std::vector<u8>(NAND_SIZE);
+  for (int i = 0; i < NAND_TOTAL_BLOCKS; i++)
+  {
+    file.ReadBytes(m_nand.data() + i * NAND_BLOCK_SIZE, NAND_BLOCK_SIZE);
+    file.Seek(NAND_ECC_BLOCK_SIZE, SEEK_CUR);  // We don't care about the ECC blocks
+  }
+  m_nand_keys = std::vector<u8>(NAND_KEYS_SIZE);
+  file.ReadBytes(m_nand_keys.data(), NAND_KEYS_SIZE);
+  return true;
+}
+
+void NANDImporter::FindSuper()
+{
+  u32 super = 0;
+  u32 newest_version = 0;
+  for (int pos = 0x1fc00000; pos < NAND_SIZE; pos += 0x40000)
+  {
+    if (!memcmp(m_nand.data() + pos, "SFFS", 4))
+    {
+      u32 version = Common::swap32(m_nand.data() + pos + 4);
+      if (super == 0 || version > newest_version)
+      {
+        super = pos;
+        newest_version = version;
+      }
+    }
+  }
+  m_nand_fat_offset = super + 0xC;
+  m_nand_fst_offset = m_nand_fat_offset + 0x10000;
+}
+
+std::string NANDImporter::GetPath(NANDFSTEntry* entry, const std::string& parent_path)
```

Comment:
`entry` can be a const pointer. `entry->name` just needs to be reinterpret_casted to a `const char*`